_ml-commons-plugin/agents-tools/mcp/mcp-connector.md

```diff
@@ -0,0 +1,253 @@
+| `url` | String | Yes | The complete base URL of the MCP server, including protocol, hostname, and port if not using the default port (for example, `https://my-mcp-server.com:8443`). |
+| `credential` | Object | Yes | Contains sensitive authentication information such as API keys or tokens. Values stored in this object can be securely referenced in the `headers` section using the `${credential.*}` syntax. |
+| `headers` | Object | No | HTTP headers to include with requests to the MCP server. For authentication headers, use the `${credential.*}` syntax to reference values from the `credential` object (e.g., `"Authorization": "Bearer ${credential.mcp_server_key}"`).  |
+
+The response contains the connector ID:
+
+```json
+{
+  "connector_id": "NZ2W2ZUBZ_3SyqdOvh2n",
+}
+```
+
+## Step 2: Register a model
+
+Register any externally hosted large language model (LLM) using a connector. For a list of supported models, see [Supported connectors]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/supported-connectors/).
+
+For example, to register an OpenAI chat model, send the following request:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "My OpenAI model: gpt-4",
+  "function_name": "remote",
+  "description": "Test model registration (this example uses OpenAI, but you can register any model)",
+  "connector": {
+    "name": "My OpenAI Connector: gpt-4",
+    "description": "Connector for the OpenAI chat model",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+      "model": "gpt-4o"
+    },
+    "credential": {
+      "openAI_key": "<YOUR_API_KEY>"
+    },
+    "actions": [
+      {
+        "action_type": "predict",
+        "method": "POST",
+        "url": "https://api.openai.com/v1/chat/completions",
+        "headers": {
+          "Authorization": "Bearer ${credential.openAI_key}"
+        },
+        "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": [{\"role\":\"developer\",\"content\":\"${parameters.system_instruction}\"},${parameters._chat_history:-}{\"role\":\"user\",\"content\":\"${parameters.prompt}\"}${parameters._interactions:-}], \"tools\": [${parameters._tools:-}],\"parallel_tool_calls\":${parameters.parallel_tool_calls},\"tool_choice\": \"${parameters.tool_choice}\" }"
+      }
+    ]
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the model ID:
+
+```json
+{
+  "task_id": "K_iQfpYBjoQOEoSHN3wU",
+  "status": "CREATED",
+  "model_id": "LPiQfpYBjoQOEoSHN3zH"
+}
+```
+
+To check the status of the operation, provide the task ID to the [Tasks API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/tasks-apis/get-task/). Once the registration is complete, the task `state` changes to `COMPLETED`.
+
+## Step 3: Register an agent for accessing MCP tools
+
+Currently, MCP tools can only be used with [_conversational_ agents]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/index/#conversational-agents) or _plan and execute_ agent types. 
```

Comment:
Is _plan and execute_ agent the same as _plan, execute, and reflect_ agent described in [this PR](https://github.com/opensearch-project/documentation-website/pull/9743/)?